src/tree/ResolvedFunctionAppResource.ts

```diff
@@ -91,18 +90,21 @@ export class ResolvedFunctionAppResource extends ResolvedFunctionAppBase impleme
         }
     }
 
-    public async getSite(context: IActionContext): Promise<ParsedSite> {
-        let site: ParsedSite | undefined = this._site;
-        if (!site) {
+    public async initSite(context: IActionContext): Promise<void> {
+        if (!this._site) {
             const webClient = await createWebSiteClient({ ...context, ...this._subscription });
-            const rawSite = await webClient.webApps.get(this.queryResult.resourceGroup, this.queryResult.name);
-            site = new ParsedSite(rawSite, this._subscription);
+            const rawSite = await webClient.webApps.get(this.dataModel.resourceGroup, this.dataModel.name);
+            this._site = new ParsedSite(rawSite, this._subscription);
             this.data = rawSite;
-            this._site = site;
             this.addValuesToMask(this._site);
         }
+    }
 
-        return site;
+    public get site(): ParsedSite {
+        if (!this._site) {
+            throw new Error(localize('siteNotSet', 'Site is not set. Ensure the site is initialized before accessing it.'));
+        }
+        return this._site;
```

Comment:
When we throw this, can we also just load the site in the background so that if a user tries again it'll work?